### ISSUE TITLE
Azure Release Pipeline : Deployment Fails

### ISSUE THREAD

**Issue by AvinandanPatra** [Azure/azure-api-management-devops-resource-kit#400]
I referred the great demo provided by Miao in he below video
https://channel9.msdn.com/Shows/Azure-Friday/Build-a-CICD-pipeline-for-API-Management-Part-2

I am getting error while deploying the Global APIM Master template with linked Template to API specific master template generated in the api specific sub folder (refer at 3:31 of the demo). 
Getting the below error :-

"code": "DeploymentActive",
"message": "Unable to edit or replace deployment 'apisTemplate': previous deployment from '5/21/2020 5:11:02 PM' is still active (expiration time is '5/28/2020 5:10:58 PM'). Please see https://aka.ms/arm-deploy for usage details."

Deploying directly from the api specific master template works, issue is only when we try to link a linked api master template from global apim master template.

Steps : 
1. Use Extract command with following extractparam settings json
{
    "sourceApimName": "ContosoApiMgmt",
    "destinationApimName": "ContosoAPIMDev2020",
    "resourceGroup": "ContosoResourceGroup",
    "fileFolder": "..\\apis",
     "linked": "true",
    "linkedTemplatesBaseUrl": "https://XXXXX/featurebuild-blob",
    "policyXMLBaseUrl": "https://XXXXX/featurebuild-blob/policies",
     "splitAPIs": **"true",**
     "paramNamedValue": "true",
     "paramApiLoggerId": "true"
}

The above step generates files in an apis Folder (as mentioned in the fileFolder param above)
apis
      
      1) petstore-api Folder
            a) Policies Folder  --> contains pet-store specific policies
            b) ContosoApiMgmt-globalServicePolicy.template.json
            c)  ContosoApiMgmt-master.template.json
            d) ContosoApiMgmt-namedValues.template.json
            e) ContosoApiMgmt-parameters.json
            f)  ContosoApiMgmt--petstore-api-api.template.json
            g) ContosoApiMgmt-products.template.json
     2) Policies Folder --> contains all policies of the APIM
     3) ContosoApiMgmt--apis.template.json
     4) ContosoApiMgmt-apiVersionSets.template.json
     5) ContosoApiMgmt-globalServicePolicy.template.json
     6) ContosoApiMgmt-master.template.json 
     7) ContosoApiMgmt-namedValues.template.json
     8) ContosoApiMgmt-parameters.json
     9) ContosoApiMgmt-products.template.json

Steps 2 : Remove all the resources in the Generated APIM Master Template (point 6 above)
Steps 3 : Add the following code snipped as done in demo at 3:31 of the demo such that point 7 above is linked to point 1c above.
`	  "resources": [
		{
		  "properties": {
			"mode": "Incremental",
			"templateLink": {
			  "uri": "[concat(parameters('LinkedTemplatesBaseUrl'), '/petstore-api/ContosoApiMgmt-master.template.json')]",
			  "contentVersion": "1.0.0.0"
			},
			"parameters": {
			  "ApimServiceName": {
				"value": "[parameters('ApimServiceName')]"
			  },
			   "LinkedTemplatesBaseUrl": {
				"value": "[concat(parameters('LinkedTemplatesBaseUrl'), '/petstore-api')]"
			  },
			  "PolicyXMLBaseUrl": {
				"value":  "[concat(parameters('LinkedTemplatesBaseUrl'), '/petstore-api/policies')]"
			  },
			  "ApiLoggerId": {
				"value": "[parameters('ApiLoggerId')]"
			  },
				  "NamedValues": {
				"value": "[parameters('NamedValues')]"
			  }
			}
		  },
		  "name": "petstore-api",
		  "type": "Microsoft.Resources/deployments",
		  "apiVersion": "2018-01-01",
		  "dependsOn": []
		}
	  ]
	}`

Step 4 : Prepare Build and Release pipeline as shown in the demo

**Comment by BryanSoltis** [comment]
I am having this same issue. I can successfully deploy the "master" template within my API folder. When I attempt to change my Release Pipeline to use "global master" template (outside of my API folder), I get an error that the "nameValuesTemplate" cannot be deployed because of existing deployment. I have verified that there are no other deployments in my resource group / subscriptions.

It appears as if the process of deploying the "global master" template starts a deployment of some sort, and when the linked api template (inside my API folder) attempts to be deployed, the ARM engine thinks the deployment is already running or something. I have tried this process within an Azure DevOps Release Pipeline, through the portal (by uploading my master template), and through the CLI *using local templates).

**Comment by BryanSoltis** [comment]
Just after posting my message, I figured out my issue. When following video guide, I used the Extractor tool to create my API ARM templates, with the master templates (specifying the the LinkedTemplatesBaseUrl value). I then used the generated "master" template for my "global master template", removing all the existing resources. I then added my new API as a resource in the "global master template", copying the resource details from the API specific one. Because of this, the name of my deployment in the "global master template" was the same as the nested one, therefore causing an issue.

 To simplify it, be sure you are properly naming the deployments in the "global" and "nested" template files to ensure they are unique.
